src/System.Reflection.Emit/tests/MethodBuilder/MethodBuilderGetGenericArguments.cs

```diff
@@ -0,0 +1,132 @@
+            string methodName = null;
+
+            methodName = TestLibrary.Generator.GetString(false, false, true, MinStringLength, MaxStringLength);
+            MethodBuilder builder = TestTypeBuilder.DefineMethod(methodName,
+                MethodAttributes.Public);
+
+            Assert.Null(builder.GetGenericArguments());
+        }
+
+        [Fact]
+        public void PosTest2()
+        {
+            string methodName = null;
+            methodName = TestLibrary.Generator.GetString(false, false, true, MinStringLength, MaxStringLength);
+            MethodBuilder builder = TestTypeBuilder.DefineMethod(methodName,
+                MethodAttributes.Public);
+            string[] typeParamNames = { "T", "U" };
+            GenericTypeParameterBuilder[] desiredParameters = builder.DefineGenericParameters(typeParamNames);
+
+            Type[] actualParameters = builder.GetGenericArguments();
+            Assert.True(VerificationHelper(desiredParameters, actualParameters));
+        }
+
+        [Fact]
+        public void PosTest3()
+        {
+            string methodName = null;
+            string paramName = null;
+            methodName = TestLibrary.Generator.GetString(false, false, true, MinStringLength, MaxStringLength);
+            paramName = TestLibrary.Generator.GetString(false, false, true, MinStringLength, MaxStringLength);
+
+            Type[] paramTypes = new Type[] { typeof(int) };
+            MethodBuilder builder = TestTypeBuilder.DefineMethod(
+                methodName,
+                MethodAttributes.Public,
+                typeof(void),
+                paramTypes);
+            string[] typeParamNames = { "T" };
+            GenericTypeParameterBuilder[] desiredParameters = builder.DefineGenericParameters(typeParamNames);
+            ParameterBuilder paramBuilder = builder.DefineParameter(
+                1,
+                ParameterAttributes.HasDefault,
+                paramName);
+
+            Type[] actualParameters = builder.GetGenericArguments();
+            Assert.True(VerificationHelper(desiredParameters, actualParameters));
+        }
+
+        [Fact]
+        public void PosTest4()
+        {
+            string methodName = null;
+            methodName = TestLibrary.Generator.GetString(false, false, true, MinStringLength, MaxStringLength);
+
+            MethodBuilder builder = TestTypeBuilder.DefineMethod(
+                methodName,
+                MethodAttributes.Public);
+            string[] typeParamNames = { "T", "U" };
+            GenericTypeParameterBuilder[] desiredParameters = builder.DefineGenericParameters(typeParamNames);
+
+            builder.SetReturnType(desiredParameters[0].AsType());
+
+            Type[] actualParameters = builder.GetGenericArguments();
+            Assert.True(VerificationHelper(desiredParameters, actualParameters));
+        }
+
+        private bool VerificationHelper(GenericTypeParameterBuilder[] desiredParameters, Type[] actualParameters)
```

Comment:
Rather than returning true or false, wouldn't it be better to Assert.True/False/Equals/etc. the various conditions and have VerificationHelper be a void-returning method?  That way if something does fail, the assert will tell us exactly what.